lib/datadog/tracing/trace_id_generation.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# typed: ignore
+
+require_relative 'utils'
+require_relative 'metadata/tagging'
+require_relative 'metadata/ext'
+
+module Datadog
+  module Tracing
+    module TraceIdGeneration
```

Comment:
Is there any reason for this to not be directly in `TraceOperation` instead on its own module?

Reply:
I recognize that trace id is generated from `SpanOperation` and `Span`, I thought extracting `TraceIdGeneration` to be included for `SpanOperation` and `Span` would be consistent.